pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -515,6 +515,21 @@ func (i *Instances) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress,
 			)
 		}
 	}
+
+	// ensure that the "primary" IP address of the machine is always included, since the above
+	// guest.net call is limited to 16 unsorted interfaces and the e.g. eth0 iface may not have
+	// been included.
+	err = getVirtualMachineManagedObjectReference(ctx, i.client, vm, "guest.ipAddress", &mvm)
+	if err != nil {
+		return nil, err
+	}
+	v1.AddToNodeAddresses(&addrs,
+		v1.NodeAddress{
+			Type:    v1.NodeInternalIP,
```

Comment:
hmmm I am not sure now but I think the external IP is the IP addresses that could be accessed from public network? Also may I ask why guest.ipAddress is guaranteed to be eth0?